### PR TITLE
Mm 1886 Add loading screen to channel invite modal

### PR DESCRIPTION
**Pull request by nickago**

### CODE REVIEW

**Comment by hmhealey** [comment]
This isn't working for me. After I open the add members dialog for the first time, it shows the same potential members every time I open the dialog, even after I switch channels. So if I open it in the town square, it'll show no members to add (which is correct), but then if I switch to a brand new channel, it'll still show no members to add.

**Comment by nickago** [comment]
That's peculiar, I'll investigate and get back to you

**Comment by nickago** [comment]
@hmhealey @jwilander Is there any particular reason why just this modal is restricted to only update while shown?

**Comment by jwilander** [comment]
I believe there was a community pull request that did that to prevent extra work being done when the modal wasn't being shown. I think it would be preferable to keep it that way and just pull from the stores once the modal is shown

File: web/react/components/channel_invite_modal.jsx [hunk 1]
```diff
@@ -4,131 +4,157 @@
 var UserStore = require('../stores/user_store.jsx');
 var ChannelStore = require('../stores/channel_store.jsx');
 var MemberList = require('./member_list.jsx');
+var LoadingScreen = require('./loading_screen.jsx');
 var utils = require('../utils/utils.jsx');
 var client = require('../utils/client.jsx');
 var AsyncClient = require('../utils/async_client.jsx');
 
 function getStateFromStores() {
```
Comment (crspeller) [inline web/react/components/channel_invite_modal.jsx]: Why is this not a member function?

**Comment by hmhealey** [comment]
Even with the state being updated on show, there's still something not working with this. I'm still having the issue where the dialog doesn't update when you switch channels (although it doesn't happen if you actually add a member on the first channel you open). There seems to be something stopping the dialog from re-rendering after you switch channels.

**Comment by nickago** [comment]
I'm not getting any of those issues, but I'll keep looking

**Comment by nickago** [comment]
Alright, I think I found the edge case, let me know if it works for you now @hmhealey

**Comment by hmhealey** [comment]
Awesome, it works. Unfortunately, it also needs a rebase now.

**Comment by crspeller** [comment]
No longer needs rebase.

**Comment by nickago** [comment]
yay

**Comment by hmhealey** [comment]
Cool, looks good then.

**Pull request merged**